### PR TITLE
#18715 : Allow users to specify the order PP Filters appear in the list.

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes

* Added an optional "sort" property that can be used to specify filter ordering. A default sort value of 1000 is used if not present. Filters with the same sort value fall back to using alphabetical sorting.
* Moved business logic from the `PushPublishFiltersInitializer` to the `PublisherAPIImpl`, where it should be.
* Moved business logic from the `PushPublishFilterResource` REST Endpoint into the `PublisherAPIImpl`, where it should be.
* Adjusted existing Integration Tests, and added new ones.

### Checklist
- [ ] Tests

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publishing/FilterDescriptor.java [hunk 1]
```diff
@@ -40,36 +46,62 @@ public class FilterDescriptor implements Comparable<FilterDescriptor> {
 
     private String key;
     private final String title;
+    private final String sort;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/FilterDescriptor.java:49]: why not the sort is an integer?

File: dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java [hunk 2]
```diff
@@ -1,40 +1,61 @@
...
+ */
 public class PublisherAPIImpl implements PublisherAPI {
 
     private final PublishAuditAPI publishAuditAPI = PublishAuditAPI.getInstance();
     private final LocalSystemEventsAPI localSystemEventsAPI = APILocator.getLocalSystemEventsAPI();
-    private final Map<String,FilterDescriptor> loadedFilters = new ConcurrentHashMap<>();
-
+    private List<FilterDescriptor> filterList = new CopyOnWriteArrayList<>();
+    /** Path where the YAML files are stored */
+    private final String PUBLISHING_FILTERS_FOLDER =
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java:56]: I am not sure if this should be a lazy or not

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publishing/FilterDescriptor.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publishing/FilterDescriptor.java:49]: Met with @jdotcms and confirmed that we're not handling integer values in our YAML files, only BOOL, STRING, or SELECT. See: https://www.dotcms.com/docs/latest/apps-integrations#FileFormat

File: dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java]: Done!

**Review by nollymar — APPROVED** [review]

**Comment by jcastro-dotcms** [comment]
Closing this PR, as it has too many conflicts with several Integration Tests.

**Pull request closed without merging**